wpilibj/build.gradle

```diff
@@ -242,6 +242,9 @@ model {
                     binary.lib project: ':hal', library: 'halAthena', linkage: 'static'
                 } else {
                     binary.lib project: ':hal', library: 'halSim', linkage: 'static'
+  		    if (binary.targetPlatform.operatingSystem.windows) {
+                        binary.linker.args "ws2_32.lib"
+                    }
```

Comment:
Similar to Thad's other comment, use `#pragma comment(lib, "Ws2_32.lib")` to pull this in in the source that uses it.  See wpiutil's hostname.cpp for an example.